### PR TITLE
WIP: Add storage.factoryReset function

### PR DESCRIPTION
**Pull request by cowlicks**
This clears all data on disk. I've found it useful for testing and development.

### CODE REVIEW

**Comment by ghostwords** [comment]
Related: #971.

File: src/js/storage.js [hunk 1]
```diff
@@ -83,6 +84,19 @@ function BadgerPen(callback) {
 }
 
 BadgerPen.prototype = {
+  /**
+   * Reset the storage on the badeger.
```
Comment (ghostwords) [inline src/js/storage.js]: Typo: badeger

**Comment by ghostwords** [comment]
The code looks simple but is actually buggy. This doesn't reset Badger settings to defaults but to blanks. This leads to JS errors (check out the options page after resetting) and probably generally incorrect behavior.

I think this is an over-optimization that adds complexity and potential bugs. When developing, why not just run `chrome.storage.local.clear()` followed by a Ctrl-R? If you are looking for local solutions to autoreload dev extensions, it appears [there are existing solutions](https://stackoverflow.com/questions/2963260/how-do-i-auto-reload-a-chrome-extension-im-developing).

**Comment by cowlicks** [comment]
stale, closing

**Comment by cowlicks** [comment]
I'd like to keep this open since there is are two related issues #1422 and #971 

I also think it could be useful for testing purposes.

**Comment by ghostwords** [comment]
Closing, as stale and low priority.

**Pull request closed without merging**